include/beman/optional26/optional.hpp

```diff
@@ -532,20 +532,20 @@ class optional {
     template <class F>
     constexpr auto transform(F&& f) & {
         using U = std::invoke_result_t<F, T&>;
-        static_assert(!std::is_array_v<U>);
-        static_assert(!std::is_same_v<U, in_place_t>);
-        static_assert(!std::is_same_v<U, nullopt_t>);
-        static_assert(std::is_object_v<U> || std::is_reference_v<U>); /// References now allowed
+        static_assert(!std::is_array_v<U>, "U must not be an array");
```

Comment:
Can you attach to PR description a log (gcc + clang) with behaviour before and after this PR?

Otherwise LGTM. I think the logs would help us to track the issues.

Do you plan to push commits with messages for all static asserts in this PR?

Reply:
Looks good to me, too.

Reply:
@neatudarius my plan is to cover as many static_assert as I can.

Reply:
> Can you attach to PR description a log (gcc + clang) with behaviour before and after this PR?
@neatudarius this is my first PR to this repo, can you elaborate what you mean by this?

Reply:
Just a cut/paste of relevant errors from the static_asserts firing.

Reply:
Do you mean running the files under `/examples` with gcc and clang and see what errors they are showing?

Reply:
> @neatudarius this is my first PR to this repo, can you elaborate what you mean by this?
Welcome @changkhothuychung to Beman! Thanks for contributions!

> Do you mean running the files under /examples with gcc and clang and see what errors they are showing?

No. So basically, try to temporary change something in the optional class on your local setup to show the compile error before and after your change. And then include such snippet with logs in PR description.

Reply:
I see, got it, will do!